content-src/components/Routes/Routes.js

```diff
@@ -1,21 +1,26 @@
 const React = require("react");
 const {Router, Route, IndexRoute} = require("react-router");
 const {createHashHistory} = require("history");
+const {connect} = require("react-redux");
+const {actions} = require("actions/action-manager");
 
 const history = createHashHistory({queryKey: false});
 
 const Routes = React.createClass({
+  onChange(data) {
+    this.props.dispatch(actions.NotifyRouteChange(data));
+  },
   render() {
     return (<Router history={history}>
       <Route path="/" component={require("components/Base/Base")}>
-        <IndexRoute title="Home" component={require("components/NewTabPage/NewTabPage")} />
+        <IndexRoute title="Home" component={require("components/NewTabPage/NewTabPage")} onEnter={data => this.onChange(data)} />
```

Comment:
Instead of this, add a listener to history (on line 7) 

```
history.listen(location => {
  this.props.dispatch....
})

```

https://github.com/mjackson/history#basic-usage